CHANGELOG.md

```diff
@@ -1,3 +1,236 @@
+- cli/cmd: cleanups part 1
+  ([#1013](https://github.com/kinvolk/lokomotive/pull/1013)).
+- test/components/kubernetes: remove kubelet pod when testing node labels
+  ([#1052](https://github.com/kinvolk/lokomotive/pull/1052)).
+- remove usage of template_file
+  ([#1046](https://github.com/kinvolk/lokomotive/pull/1046)).
+- test/*: de-duplicate value timeout and retryInterval
+  ([#1049](https://github.com/kinvolk/lokomotive/pull/1049)).
+- packet: Read BGP peer address from metadata service
+  ([#1010](https://github.com/kinvolk/lokomotive/pull/1010)).
+- pkg/assets: cleanup exported API
+  ([#936](https://github.com/kinvolk/lokomotive/pull/936)).
+- cobra upgrade to v1.1.1
+  ([#1082](https://github.com/kinvolk/lokomotive/pull/1082)),
+  ([#1091](https://github.com/kinvolk/lokomotive/pull/1091)).
+- Update grafana and other prometheus operator sub components
+  ([#963](https://github.com/kinvolk/lokomotive/pull/963)).
+- e2e: Add test for prometheus-operator PVC
+  ([#1087](https://github.com/kinvolk/lokomotive/pull/1087)).
+- cli/cmd: cleanups part 2
+  ([#1015](https://github.com/kinvolk/lokomotive/pull/1015)).
+- Add github actions
+  ([#1074](https://github.com/kinvolk/lokomotive/pull/1074)).
+- Makefile: use latest Go when building in Docker
+  ([#1083](https://github.com/kinvolk/lokomotive/pull/1083)).
+- Rename Headlamp to Web UI and add OIDC support
+  ([#1100](https://github.com/kinvolk/lokomotive/pull/1100)).
+- cli/cmd: cleanups part 3
+  ([#1018](https://github.com/kinvolk/lokomotive/pull/1018)).
+
+### Upgrading from v0.4.1
+
+#### Configuration syntax changes
+
+There have been some minor changes to the configurations of worker nodes, `labels` and `taints` format changed from `string` to `map(string)` for aws and packet platform.
+
+old:
+
+```
+labels = "testing=true"
+
+taints = "nodeType=storage:NoSchedule"
+```
+
+new:
+
+```
+labels = {
+  "testing" = "true"
+}
+
+taints = {
+  "nodeType" = "storage:NoSchedule"
+}
+```
+
+#### Cluster upgrade steps
+
+Due to Terraform and Kubernetes upgrades, there are some additional steps that you need to follow. Go to your cluster's directory and follow these steps:
+
+1. Upgrade to terraform 0.13. Follow [this guide](https://www.terraform.io/upgrade-guides/0-13.html).
+
+2. `export PLATFORM="packet"` # specify your platform.
+
+3. `rm assets/terraform-modules/$PLATFORM/flatcar-linux/kubernetes/require.tf assets/terraform-modules/$PLATFORM/flatcar-linux/kubernetes/workers/require.tf assets/terraform-modules/dns/route53/require.tf`
+
+4. `cd assets/terraform` # go to terraform directory.
+
+5. `terraform state replace-provider registry.terraform.io/-/ct registry.terraform.io/poseidon/ct` then enter yes for confirmation.
+
+6. `terraform state replace-provider registry.terraform.io/-/template registry.terraform.io/hashicorp/template` then enter yes for confirmation.
+
+7. `kubectl delete rs kube-apiserver-tab tab` # NOTE: **delete Replica Set kube-apiserver only for single node cluster**.
```

Comment:
I would remove all but the last one maybe? I think this should be more robust.

Reply:
> `I would remove all but the last one maybe`

I am sorry, but I don't quite understand what you mean by

Reply:
We shouldn't remove all replicasets for kube-apiserver, but only the outdated one, to avoid descheduling the up-to-date one. This way, upgrade process should be safer and faster, as it won't rely on checkpointer.

Reply:
For a single node cluster, isn't only one `rs` running?

Reply:
We run Single deployment, but it will concist of multiple replica sets while upgrading and with older versions.

Reply:
Aaah, got it what you meant